### PR TITLE
Enable strict types mode

### PR DESCRIPTION
**Pull request by RDIL**
Slowly but surely enabling strict null checking on the codebase for enhanced security.

Opening this PR just to note for myself where potentially breaking changes are.

### CODE REVIEW

File: components/controller.ts [hunk 1]
```diff
@@ -509,14 +513,7 @@ export class Controller {
...
 
-        if (this.contracts.has(this._pubIdToContractId.get(pubId)!)) {
-            return (
-                this.contracts.get(this._pubIdToContractId.get(pubId)!) ||
-                undefined
-            )
-        }
-
-        return undefined
+        return this.contracts.get(this._pubIdToContractId.get(pubId)!)
```
Comment (RDIL) [inline components/controller.ts:516]: these are equivalent due to .get returning undefined if not present

File: components/flags.ts [hunk 2]
```diff
@@ -152,32 +151,22 @@ const makeFlagsIni = (
...
-            LogLevel.WARN,
-            "The flags file (flags.json5) has been revamped in the latest Peacock version, and we had to remove your settings.",
-        )
-        log(
-            LogLevel.INFO,
-            "You can take a look at the new options.ini file, which includes descriptions and more!",
-        )
-
-        unlinkSync(OLD_FLAGS_FILE)
-    }
```
Comment (RDIL) [inline components/flags.ts:175]: Legacy code removal

File: components/index.ts [hunk 3]
```diff
@@ -496,25 +508,12 @@ program.description(
...
-const PEECOCK_ART = picocolors.yellow(`
- ███████████  ██████████ ██████████   █████████     ███████      █████████  █████   ████
-░░███░░░░░███░░███░░░░░█░░███░░░░░█  ███░░░░░███  ███░░░░░███   ███░░░░░███░░███   ███░
- ░███    ░███ ░███  █ ░  ░███  █ ░  ███     ░░░  ███     ░░███ ███     ░░░  ░███  ███
- ░██████████  ░██████    ░██████   ░███         ░███      ░███░███          ░███████
- ░███░░░░░░   ░███░░█    ░███░░█   ░███         ░███      ░███░███          ░███░░███
- ░███         ░███ ░   █ ░███ ░   █░░███     ███░░███     ███ ░░███     ███ ░███ ░░███
- █████        ██████████ ██████████ ░░█████████  ░░░███████░   ░░█████████  █████ ░░████
-░░░░░        ░░░░░░░░░░ ░░░░░░░░░░   ░░░░░░░░░     ░░░░░░░      ░░░░░░░░░  ░░░░░   ░░░░
-`)
```
Comment (RDIL) [inline components/index.ts:508]: we're professional now

**Comment by moonysolari** [comment]
I have a feeling that this PR has brought about more bugs than we realize...

**Comment by RDIL** [comment]
It probably has, but without people able to test it, we won't know.

**Comment by LennardF1989** [comment]
> I have a feeling that this PR has brought about more bugs than we realize...
A small price to pay for the greater good, IMO.

**Pull request merged**